src/Keboola/DbExtractor/Extractor/Extractor.php

```diff
@@ -182,7 +182,12 @@ public function export(array $table): array
         }
 
         try {
-            $result = $this->writeToCsv($stmt, $csv, $isAdvancedQuery);
+            $result = $this->writeToCsv($stmt, $csvWriter, $isAdvancedQuery);
+
+            if ($result['rows'] === 0) {
+                var_dump($this->getOutputFilename($outputTable));
+                unlink($this->getOutputFilename($outputTable));
```

Comment:
Is this a BC break?

Reply:
Not really, old `CsvFile` worked differently - it created file when it wrote first row. `CsvWriter` creates empty `.csv` file vhen is instanciated.

Reply:
Ah, right, I remember that conversation now:).  ok 👍